### PR TITLE
[FEAT] add ftx to swap

### PR DESCRIPTION
**Pull request by chabroA**
### ❓ Context

- **Impacted projects**: [`live-desktop`, `@ledgerhq/live-common`]
  
- **Linked resource(s)**:
   - https://ledgerhq.atlassian.net/browse/LIVE-806 

Adds FTX as provider in swap feature

### ☑️  TODO (before merge)

- [x] Remove `SWAP_API_BASE` staging URL from `apps/ledger-live-desktop/src/env.js`

### ✅ Checklist

- [x] **Test coverage**: _Did you write any tests to cover the changes introduced by this pull request?_
- [ ] **Atomic delivery**: _Is this pull request standalone? In order words, does it depend on nothing else?_

It depends on an update of the Swap API, cf. https://ledgerhq.atlassian.net/browse/SRE-315 

- [ ] **Breaking changes**: _Does this pull request contain breaking changes of any kind? If so, please explain why._

### 📸 Demo

Screenshot of updated parts bellow 👇 

<details>
 <summary>Settings page</summary>

</details>

<details>
 <summary>Swap page</summary>

### After:

</details>

<details>
 <summary>New widgets</summary>

Screenshot 2022-03-04 at 17 15 15
Screenshot 2022-03-04 at 17 15 47

</details>

<details>
 <summary>New error / informative messages</summary>

Screenshot 2022-06-15 at 12 47 15

Generic error:
Screenshot 2022-06-15 at 12 56 03

</details>

### 🧪  QA steps:

To put it shortly, test the different login / KYC flows for swap with FTX and FTX US.
FTX should always be first provider available.

You can use [these test credentials](https://ledgerhq.atlassian.net/wiki/spaces/WALLETCO/pages/3548938708/SWP-FTX-INTEGRATION+-+Tech+team+test+accounts) in order to test each flow. Follow the instruction on the page to use a test account.
⚠️ **Do not try to update KYC (i.e: submit the form) or enable MFA on any of these test accounts** ⚠️  

### 🚀 Expectations to reach

_Please make sure you follow these [**Important Steps**](https://github.com/LedgerHQ/ledger-live/blob/develop/CONTRIBUTING.md#important-steps)._

_Pull Requests must pass the CI and be internally validated in order to be merged._

### CODE REVIEW

File: apps/ledger-live-desktop/src/main/index.js [hunk 1]
```diff
@@ -75,6 +75,23 @@ app.on("ready", async () => {
     await installExtensions();
   }
 
+  /**
+   * Clears the session’s HTTP cache
+   * Used to remove third party cached auth tokens, among other things
+   */
+  ipcMain.handle("clearStorageData", () => {
```
Comment (ggilchrist-ledger) [inline apps/ledger-live-desktop/src/main/index.js:82]: Does this get cleared on every startup of the app?

Comment (chabroA) [reply apps/ledger-live-desktop/src/main/index.js:82]: No, only when the user "reset third party data" in the settings

File: apps/ledger-live-desktop/tests/specs/devmode.spec.ts [hunk 2]
```diff
@@ -1,11 +1,11 @@
...
 import { expect } from "@playwright/test";
-import { Layout } from "../models/Layout";
-import { Modal } from "../models/Modal";
+import test from "../fixtures/common";
 import { AccountsPage } from "../models/AccountsPage";
 import { AddAccountModal } from "../models/AddAccountModal";
-import { SettingsPage } from "../models/SettingsPage";
 import { DeviceAction } from "../models/DeviceAction";
+import { Layout } from "../models/Layout";
+import { Modal } from "../models/Modal";
```
Comment (Justkant) [inline apps/ledger-live-desktop/tests/specs/devmode.spec.ts:7]: We can revert this changes IMO

File: apps/ledger-live-desktop/tests/specs/onboarding/device-ready.spec.ts [hunk 3]
```diff
@@ -1,7 +1,7 @@
-import test from "../../fixtures/common";
 import { expect } from "@playwright/test";
-import { OnboardingPage } from "../../models/OnboardingPage";
+import test from "../../fixtures/common";
 import { DeviceAction } from "../../models/DeviceAction";
```
Comment (Justkant) [inline apps/ledger-live-desktop/tests/specs/onboarding/device-ready.spec.ts:3]: We can revert this changes IMO

File: apps/ledger-live-desktop/tests/specs/onboarding/new-device.spec.ts [hunk 4]
```diff
@@ -1,7 +1,7 @@
-import test from "../../fixtures/common";
 import { expect } from "@playwright/test";
-import { OnboardingPage } from "../../models/OnboardingPage";
+import test from "../../fixtures/common";
 import { DeviceAction } from "../../models/DeviceAction";
+import { OnboardingPage } from "../../models/OnboardingPage";
```
Comment (Justkant) [inline apps/ledger-live-desktop/tests/specs/onboarding/new-device.spec.ts:4]: We can also revert this changes IMO

**Review by Justkant — COMMENTED** [review]
Only small stuff and some questions

File: apps/ledger-live-desktop/src/env.js [hunk 5]
```diff
@@ -2,5 +2,7 @@ import "dotenv/config";
 
 process.env = {
   ...process.env,
+  // FIXME: needed to build a workable test bundle of the app, to be removed one backend API is in prod
+  SWAP_API_BASE: "https://swap.aws.stg.ldg-tech.com/v3",
```
[the remark this answers is not in the document]
Comment (chabroA) [reply apps/ledger-live-desktop/src/env.js:6]: It will stay until the Swap API is deployed in Prod (cf. https://ledgerhq.atlassian.net/browse/SRE-315)

It should be removed before being merged

[the remark this answers is not in the document]
Comment (gre) [reply apps/ledger-live-desktop/src/env.js:6]: mmh can we use `live-common` env instead? I'm not sure this env approach will work properly across all envs (internal, renderer, main), even if you only need it at one place, i would recommend we move to the unified approach of env.ts because in that way we can see in one glance what are all our endpoints 🙏

[the remark this answers is not in the document]
Comment (chabroA) [reply apps/ledger-live-desktop/src/env.js:6]: @gre indeed this `SWAP_API_BASE` is already in `env.ts` in `live-common`
Actually it was put here as well to be able to produce builds on the CI suing the staging env.

After reflexion, it can actually be removed now, the only issue is that the builds coming from the CI will not work "out of the box" until the API is deployed to prod (should happen today).
For testing the build, you will need to start the app with the above env var set manually

File: apps/ledger-live-desktop/src/config/urls.js [hunk 6]
```diff
@@ -178,6 +178,18 @@ export const urls = {
...
+        main: "https://ftx.com",
+        tos: "",
+        support: "https://ftx.com/support",
+        kyc: "",
+      },
+      ftxus: {
+        main: "https://ftx.us/",
+        tos: "",
+        support: "https://help.ftx.us/hc/en-us",
+        kyc: "",
```
[the remark this answers is not in the document]
Comment (chabroA) [reply apps/ledger-live-desktop/src/config/urls.js:191]: Actually no, will remove 👍

File: apps/ledger-live-desktop/src/env.js (hunk 5 above)
[the remark this answers is not in the document]
Comment (chabroA) [reply apps/ledger-live-desktop/src/env.js:6]: Will remove

File: apps/ledger-live-desktop/src/renderer/modals/Swap/steps/StepDevice.js [hunk 7]
```diff
@@ -133,9 +133,9 @@ const StepDevice = ({
       key={"initSwap"}
       action={action2}
       request={request}
-      onResult={({ initSwapResult, initSwapError, ...rest }) => {
-        if (initSwapError) {
-          onError(initSwapError);
+      onResult={({ initSwapResult, initSwapErrorResult, ...rest }) => {
+        if (initSwapErrorResult) {
+          onError(initSwapErrorResult);
```
[the remark this answers is not in the document]
Comment (chabroA) [reply apps/ledger-live-desktop/src/renderer/modals/Swap/steps/StepDevice.js:138]: This is actually dead code, will remove it 👍

**Review by Justkant — APPROVED** [review]
LGTM
